### PR TITLE
fix problems with the dns resolving cache

### PR DESCRIPTION
**Pull request by HejVkt**
There are problems with connecting to the memcash server by DNS name when changing the IP address of the server.

Due to the fact that the IP address is cached and there is no mechanism for resolving a new address (updating the cache), we get constant errors connecting to the server.

I have added some solution.

### CODE REVIEW

**Comment by HejVkt** [comment]
@googlebot I fixed it.

**Comment by HejVkt** [comment]
@googlebot I fixed it.

File: memcache/selector.go [hunk 1]
```diff
@@ -33,6 +33,8 @@ type ServerSelector interface {
 	// should be shared onto.
 	PickServer(key string) (net.Addr, error)
 	Each(func(net.Addr) error) error
+	// Reresolve try to reresolve address for given ip
+	Reresolve(addr *staticAddr) error
```
Comment (bradfitz) [inline memcache/selector.go:37]: extending this interfaces breaks compatibility for everybody who's already implemented this interface. 

I don't think this is workable as-is.

**Review by bradfitz — CHANGES_REQUESTED** [review]
extending this interfaces breaks compatibility for everybody who's already implemented this interface.

I don't think this is workable as-is.

**Pull request closed without merging**